griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,302 @@
+
+            docs_service = self._build_client(self.DOCS_SCOPES, "docs", "v1")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "drive", "v3")
+
+            body = {
+                "title": file_path,
+            }
+
+            doc = docs_service.documents().create(body=body).execute()
+            doc_id = doc["documentId"]
+
+            if folder_path.lower() != self.DEFAULT_FOLDER_PATH:
+                folder_id = self._path_to_file_id(drive_service, folder_path)
+                if folder_id:
+                    drive_service.files().update(fileId=doc_id, addParents=folder_id, fields="id, parents").execute()
+                else:
+                    return ErrorArtifact(f"Error: Folder not found for path {folder_path}")
+
+            if content:
+                save_content_params = {"document_id": doc_id, "content": content}
+                saved_document_id = self._save_to_doc(save_content_params)
+                return InfoArtifact(f"Content has been successfully saved to Google Doc with ID: {saved_document_id}.")
+            else:
+                return InfoArtifact(f"Google Doc '{file_path}' created with ID: {doc_id}")
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"Error creating/saving Google Doc: {e}")
+
+    @activity(
+        config={
+            "description": "Can be used to load content from memory and save it to a new Google Doc "
+                           "in the specified folder.",
+            "schema": Schema(
+                {
+                    "memory_name": str,
+                    "artifact_namespace": str,
+                    "file_name": str,
+                    Optional(
+                        "folder_path",
+                        description="Path of the folder where the file should be saved.",
+                        default=DEFAULT_FOLDER_PATH,
+                    ): str,
+                }
+            ),
+        }
+    )
+    def save_memory_artifacts_to_google_docs(self, params: dict) -> ErrorArtifact | InfoArtifact:
+        values = params["values"]
+        memory = self.find_input_memory(values["memory_name"])
+    
+        if memory:
+            artifacts = memory.load_artifacts(values["artifact_namespace"])
+    
+            if artifacts:
+                try:
+                    file_path = values["file_name"]
+                    content = "\n".join([a.value for a in artifacts])
+
+                    save_params = {
+                        "file_path": file_path,
+                        "content": content,
+                        "folder_path": values.get("folder_path", self.DEFAULT_FOLDER_PATH),
+                    }
+
+                    return self.save_content_to_google_doc(save_params)
+    
+                except Exception as e:
+                    return ErrorArtifact(f"Error: {e}")
+    
+            else:
+                return ErrorArtifact("No artifacts found.")
```

Comment:
Lowercase, remove punctuation to be consistent with other tool error messages. `no artifacts found`

Reply:
Done!